Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -12,19 +12,34 @@ public Pawn(Player player)
         public override IEnumerable<Square> GetAvailableMoves(Board board) {
             List<Square> available = new List<Square>();
             Square here = board.FindPiece(this);
-            if (Player == Player.White) {
-                if (here.Row != 0) {
-                    available.Add(new Square(here.Row - 1, here.Col ));
-                }
-                if (here.Row == 6) {
-                    available.Add(new Square(4, here.Col ));
+            if (Player == Player.White)
+            {
+                Square targetMove = new Square(here.Row - 1, here.Col);
+                if (targetMove.IsValid() && board.IsSquareEmpty(targetMove))
+                {
+                    Square targetMove2 = new Square(here.Row - 2, here.Col);
+                    if (here.Row != 0)
+                    {
+                        available.Add(targetMove);
+                    }
+                    if (here.Row == GameSettings.BoardSize - 2 && board.IsSquareEmpty(targetMove2))
+                    {
+                        available.Add(targetMove2);
+                    }
                 }
             } else {
-                if (here.Row != 7) {
-                    available.Add(new Square(here.Row + 1, here.Col));
-                }
-                if (here.Row == 1) {
-                    available.Add(new Square(3, here.Col ));
+                Square targetMove = new Square(here.Row +1, here.Col);
+                if (targetMove.IsValid() && board.IsSquareEmpty(targetMove))
+                {
+                    Square targetMove2 = new Square(here.Row +2, here.Col);   
```

Comment:
Missing space after + sign.
You can set up rider formatter to run automatically when you save the file.